app/routes/goal_routes.py

```diff
@@ -1 +1,119 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from app.models.goal import Goal  
+from ..db import db
+from datetime import datetime, timezone
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+##refactor validate!! 
+@goals_bp.post("/<goal_id>/tasks")
+def create_goal_with_task(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    request_body = request.get_json()
+    request_body["goal_id"] = goal.id
+
+    try:
+        new_task = Task.from_dict(request_body)
+    except KeyError as error:
+        response = {"message": f"Invalid requests: missing {error.ergs[0]}"}
+        abort(make_response(response, 400))
```

Comment:
This logic looks very similar to the logic we wrote in our create routes, what refactoring could be done here to move this code out of our routes into a helper function to support maintainability and scalability? (Hint: We did a refactor like this in Flasky).